Framework/MDAlgorithms/src/IntegratePeaksMDHKL.cpp

```diff
@@ -0,0 +1,319 @@
+  double errSqSum = 0.0;
+  double measuredErrSqSum = 0.0;
+  for (int Hindex = 0; Hindex < gridPts[0]; Hindex++) {
+    for (int Kindex = 0; Kindex < gridPts[1]; Kindex++) {
+      for (int Lindex = 0; Lindex < gridPts[2]; Lindex++) {
+        int iHKL = Hindex + gridPts[0] * (Kindex + gridPts[1] * Lindex);
+        if (!boost::math::isnan(F[iHKL]) && !boost::math::isinf(F[iHKL])) {
+          measuredPoints = measuredPoints + 1;
+          measuredSum = measuredSum + F[iHKL];
+          measuredErrSqSum = measuredErrSqSum + SqError[iHKL];
+          if (F[iHKL] > minIntensity) {
+            int neighborPoints = 0;
+            for (int Hj = -2; Hj < 3; Hj++) {
+              for (int Kj = -2; Kj < 3; Kj++) {
+                for (int Lj = -2; Lj < 3; Lj++) {
+                  int jHKL =
+                      Hindex + Hj +
+                      gridPts[0] * (Kindex + Kj + gridPts[1] * (Lindex + Lj));
+                  if (Lindex + Lj >= 0 && Lindex + Lj < gridPts[2] &&
+                      Kindex + Kj >= 0 && Kindex + Kj < gridPts[1] &&
+                      Hindex + Hj >= 0 && Hindex + Hj < gridPts[0] &&
+                      F[jHKL] > minIntensity) {
+                    neighborPoints = neighborPoints + 1;
+                  }
+                }
+              }
+            }
+            if (neighborPoints >= neighborPts) {
+              peakPoints = peakPoints + 1;
+              peakSum = peakSum + F[iHKL];
+              errSqSum = errSqSum + SqError[iHKL];
+            }
+          }
+        } else {
+          double minR =
+              sqrt(std::pow(float(Hindex) / float(gridPts[0]) - 0.5, 2) +
+                   std::pow(float(Kindex) / float(gridPts[1]) - 0.5, 2) +
+                   std::pow(float(Lindex) / float(gridPts[0]) - 0.5, 2));
+          if (minR < 0.05) {
+            intensity = 0.0;
+            errorSquared = 0.0;
+            return;
+          }
+        }
+      }
+    }
+  }
+  double ratio = float(peakPoints) / float(measuredPoints - peakPoints);
+  intensity = peakSum - ratio * (measuredSum - peakSum);
+  errorSquared = errSqSum + ratio * (measuredErrSqSum - errSqSum);
+  return;
+}
+
+/**
+ * Runs the BinMD algorithm on the input to provide the output workspace
+ * All slicing algorithm properties are passed along
+ * @return MDHistoWorkspace as a result of the binning
+ */
+MDHistoWorkspace_sptr IntegratePeaksMDHKL::binEvent(int h, int k, int l,
+                                                    double box, int gridPts,
+                                                    IMDWorkspace_sptr ws) {
+  IAlgorithm_sptr binMD = createChildAlgorithm("BinMD", 0.0, 0.3);
+  binMD->setProperty("InputWorkspace", ws);
+  binMD->setProperty("AlignedDim0",
+                     "[H,0,0]," + boost::lexical_cast<std::string>(h - box) +
+                         "," + boost::lexical_cast<std::string>(h + box) + "," +
+                         boost::lexical_cast<std::string>(gridPts));
```

Comment:
Have you considered `std::to_string`? It should work for `gridPts`, but you may be better off using boost::lexical_cast with floating point numbers.